daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -384,6 +384,27 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
       case (acc, _) => acc
     } -- localContracts.keySet
 
+  /** Return all the contract keys referenced by this transaction.
+    * This includes the keys created, exercised, fetched, or looked up, even those
+    * that refer transient contracts or that appear under a rollback node.
+    */
+  final def contractKeys(implicit
+      ev: HasTxNodes[Nid, Cid] <:< HasTxNodes[_, Value.ContractId]
```

Comment:
What is an `ev`? Can you use meaningful parameter names, please?

Reply:
`ev` is pretty conventional in Scala-ese for "evidence".

I prefer the latter, but I think it's quite common (kind of like "i" for an iterator or "e" for an exception).

Reply:
https://github.com/digital-asset/daml/pull/9442

Reply:
Thanks for the explanation, @SamirTalwar-DA.

Reply:
The method [Iterable#toMap](https://github.com/scala/scala/blob/8684ae833dcfeac6107343fcca5501301e509eef/src/library/scala/collection/GenTraversableOnce.scala#L648) is the standard example of method using such implicit (and syntax)